### ISSUE TITLE
[GENERAL SUPPORT]: SEBO with upperboundarycondition

### ISSUE THREAD

**Issue by souravdey94** [facebook/Ax#3514]
### Question

I have been using qNoisyExpectedHypervolumeImprovement with SEBO for sparse (L0) optimization. But I need to more exploration. I want to try UpperBoundaryCondition and a few other acquisition functions. However, those are not in the Ax registry. I am in general referring to the tutorial at https://ax.dev/docs/tutorials/modular_botax/ and taking the qUCB constructor from https://github.com/pytorch/botorch/blob/main/botorch/acquisition/input_constructors.py#L477. 
But it doesn't have any parameter from 'sparsity threshold' or 'penalty' which are required fro SEBO.

Is it possible to use UCB and other acquisition functions with SEBO?

### Please provide any relevant code snippet if applicable.

```shell

```

### Code of Conduct

- [x] I agree to follow this Ax's Code of Conduct

**Comment by mgrange1998** [comment]
Hi, thanks for opening this issue!
It may be possible to use UCB, tagging @sdaulton who can speak to this in more detail. 



In the meantime, could you provide more details about your use case? Specifically
1. A code snipped with your set up you've been using to test qNoisyExpectedHypervolumeImprovement with SEBO for sparse (L0) optimization
2. Details about "needing more exploration" and why qUCB is required

**Comment by souravdey94** [comment]
The  generation strategy and experiment setup is attached here.


target_point_minus1 = torch.tensor([0 for _ in range(length_minus1)], **tkwargs)
    print('target_point_minus1', target_point_minus1)

def gen_strategy(target_point_minus1,length,SURROGATE_CLASS):

    gs = GenerationStrategy(
    steps=[
        # GenerationStep(
        #     model=Models.SOBOL,
        #     num_trials=15,  # https://github.com/facebook/Ax/issues/922
        # # #     min_trials_observed=3,
        # # #     # max_parallelism=20,
        # # #     model_kwargs={"seed": 999},
        # # #     model_gen_kwargs={},
        # ),
        GenerationStep(
            model=Models.BOTORCH_MODULAR,
            num_trials=-1,  # No limitation on how many trials should be produced from this step
            #max_parallelism=5,  # Parallelism limit for this step, often lower than for Sobol
            # More on parallelism vs. required samples in BayesOpt:
            # https://ax.dev/docs/bayesopt.html#tradeoff-between-parallelism-and-total-number-of-trials
            should_deduplicate=True,
            model_kwargs={  # Kwargs to pass to `BoTorchModel.__init__`
                "surrogate": Surrogate(botorch_model_class=SURROGATE_CLASS),
                "acquisition_class": SEBOAcquisition,
                 "botorch_acqf_class": qNoisyExpectedHypervolumeImprovement,
                "acquisition_options": {
                    "penalty": "L0_norm", # it can be L0_norm or L1_norm.
                    "target_point": target_point_minus1, 
                    "sparsity_threshold": 10,
                },
            },     
        )
    ]
)

def setup_experiment(solvent_names_minus1,range_min_minus1,range_max_minus1,gs):

    ax_client = AxClient(generation_strategy=gs)

    print('len(solvent_names_minus1) in setup_exp', len(solvent_names_minus1))

    sum_str = ""
    for index, name in enumerate(solvent_names_minus1):
        # print('index', index, 'name', name,'sum_str', sum_str)
        if index == 0:
            sum_str = str(name) + " " + "+"
        elif index < len(solvent_names_minus1) - 1:
            sum_str = sum_str + " " + str(name) + " " + "+"
        else:
            sum_str = sum_str + " " + str(name)
 
    sum_str = sum_str + " <= 1.00"

    print('sum_str', sum_str)

    # sum_str_1 = "solvent1 + solvent2 + solvent3 + solvent4 + solvent5 + solvent6 >= 1.00"
    # sum_str_2 = "solvent1 + solvent2 + solvent3 + solvent4 + solvent5 + solvent6 <= 1.01"

    ax_client.create_experiment(
        name="solventproject",
 
        
        parameters=[
            {
                "name": solvent_names_minus1[i],
                "type": "range",
                "bounds": [float(range_min_minus1[i]), float(range_max_minus1[i])],
                "value_type": "float",  # Optional, defaults to inference from type of "bounds".
                "log_scale": False,  # Optional, defaults to False.
            }
            for i in range(len(solvent_names_minus1))
        ],
        objectives={"blend_score": ObjectiveProperties(minimize=False)},
        parameter_constraints=[sum_str],  # Optional.
        outcome_constraints=["lnorm <= 1.02","lnorm >= 0.98",],  # Optional.

    )

    return ax_client


parameterizations, optimization_complete = ax_client.get_next_trials(batch_size)
    print('optimization_complete', optimization_complete)
    print('parameterizations', parameterizations)
    for trial_index, parameterization in list(parameterizations.items()):
        print('parameterization in run_optimization before', parameterization)

        #Code inserted here

        param_sum_minus1 = 0
        for keys in parameterization:
                param_sum_minus1 += parameterization[keys]
        print('param_sum_minus1', param_sum_minus1)
        parameter_last = 1.00 - param_sum_minus1


        parameterization[solvent_names[len(solvent_names)-1]] = parameter_last
        print('parameterization in run_optimization after', parameterization)



For the question. "Details about "needing more exploration" and why qUCB is required"

Our parameter space is 36-d long and we are using calibrating our system against different 'sparsity_threshold'. Our target post is [0]-36d.  We want to find maxima value on the Pareto frontier with 5-6 non-zero parameters. Depending on our stating point, we are getting stuck at local maxima. I didn't find any tunable parameter for qNoisyExpectedHypervolumeImprovement to increase the exploration. qUCB has the parameter beta that can be tuned to increase exploration. We wanted to use that with L0 regularisation.

**Comment by sdaulton** [comment]
Did you trying `qLogNoisyExpectedHypervolumeImprovement`? That acquisition function should be easier to optimize and will likely lead to better results.

Regarding using UCB, Ax only supports using treating sparse optimization as a multi-objective optimization problem (Sec 4.1 in https://arxiv.org/pdf/2203.01900).

**Comment by lena-kashtelyan** [comment]
This looks inactive; @souravdey94 please reopen this issue if you do follow-up! We might not see a comment on a closed issue.
